CMakeLists.txt

```diff
@@ -252,6 +252,11 @@ set(
   SPDLOG_NO_TLS
 )
 
+if (CMAKE_CXX_COMPILER_ID MATCHES ".*Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "15.0" AND NOT IRODS_BUILD_AGAINST_LIBCXX)
+  message(AUTHOR_WARNING "Adding `consteval=constexpr` to `IRODS_COMPILE_DEFINITIONS_PRIVATE` to work around Clang bug with libstdc++ 13.")
+  list(APPEND IRODS_COMPILE_DEFINITIONS_PRIVATE "consteval=constexpr")
```

Comment:
I don't think we use `consteval` at all. The problematic usages of `consteval` aren't in our code, but in libstdc++'s headers.

Reply:
I'm going to open an issue to revisit this. It can probably be a bit more surgical, and I'd like to look into other solutions as well. But I think for the purposes of landing Ubuntu 24.04 support, this is fine for now.

Reply:
Agreed. An issue to revisit this is good. Especially once `consteval` is properly supported by compilers. Until then, iRODS code must avoid use of that keyword.

Reply:
#7782